Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,31 @@
+# create a lock using `flock (2)`. A name is required as first argument.
+# the lock will be automatically unlocked when the shell process quits.
+lock() {
+  local name="$1"
+  local lock_file="/tmp/homebrew-${name}.lock"
+  # open the lock file to file descriptor 3, so the shell process can hold the lock.
```

Comment:
OK, I tried to use a variable. However, Bash 3 doesn't support to use variable for FD.